limiter/limiter.go

```diff
@@ -0,0 +1,54 @@
+package limiter
+
+import "time"
+
+type LeakyBucket struct {
+	bucket chan struct{}
+	quit   chan struct{}
+}
+
+func (rl *LeakyBucket) fill(burst int) {
+	for i := 0; i < burst; i++ {
+		rl.bucket <- struct{}{}
+	}
+}
+
+func (rl *LeakyBucket) drip(drip float64) {
+	driptime := float64(1000000000) / drip
```

Comment:
Yes, OK that makes sense. Replacing the `100000000` with `time.Second` and immediately casting the result of the division to a `time.Duration` will make it clearer.

Also perhaps calling the argument `dripFrequency` and the result `dripPeriod` will make the relationship clearer